### ISSUE TITLE
Sign Up page : Make the signup feature work

### ISSUE THREAD

**Issue by joshi1983** [hhaccessibility/hhaccessibility.github.io#58]
Don't make it pretty since we're still waiting on UX sketches to be finalized but make it work for first name, last name, password...

The emailing step may not work in all hosting environments.

The following is how the login process should work:
1. Fill in form including first name, last name, email address, password, confirmed password and captcha.
1. User clicks "Sign up".
1. If anything is invalid, show messages and if all is good, create the account and email an activation email.
1. User clicks an activation link in the email to go back to the website and activate the account.
1. User can then sign in and update his profile.

**Tasks** 
- [x] validation checks for things like the retyped password matching the original password
- [x] validation checks for non-empty first name and last name
- [x] validation checks for email address not being used yet
- [x] validation checks for email containing an @ sign
- [x] show a message when successfully created an account
- [x] email the user to confirm the email address is valid.
- [x] show message to user pushing them to check email if he attempts to log into the new account before the email activation step is completed

This is related to #199.

**Comment by joshi1983** [comment]
Everything I did on this is merged.  There is a test account people can sign in with.  Some changes should likely be made to switch from a username to an email address.  I'm unassigning myself because I won't be working more on this for a little while.  @areebakamal will be working on #62 so this issue should probably be avoided for a bit to minimize toe stomping from parallel work on the same files.

**Comment by joshi1983** [comment]
The signup is partly working.  
It still needs some things like: 
- validation checks for things like the retyped password matching the original password
- validation checks for non-empty first name and last name
- validation checks for email address not being used yet
- validation checks for email containing an @ sign
- more testing since I had trouble logging into newly created accounts on app.accesslocator.com

**Comment by joshi1983** [comment]
The validation checks are working.  I'm able to sign into the account after creating it.

More things to do include:
- show a message when successfully created an account
- email the user to confirm the email address is valid.
- show message to user pushing them to check email if he attempts to log into the new account before the email activation step is completed

**Comment by Blandine-AA** [comment]
This is the updated Sign Up page 
![bfl-4 signuppage](https://cloud.githubusercontent.com/assets/20052162/21229634/47ae06da-c2b0-11e6-8b0b-4dc2631da880.jpg)

I think we should simplify this some more and remove the First name and Last name. the user can fill this when creating his profile. Let's wait and see what feedback we get from the first round of testing.

**Comment by stephanosm** [comment]
You can't separate them because later in the profile both names will appear either in the First name or Last name field, thus causing more problems.

**Comment by Blandine-AA** [comment]
When creating an/my account the password fields are pre-populated. Why? I think they should be left blank, since it is a new account.

**Comment by joshi1983** [comment]
@Blandine-AA what do you mean by "pre-populated"?  There is no password given by default by our application.  If you're new to the sign up page and seeing the password prepopulated, I strongly suspect it'll be a stored password in your web browser.  

It is blank until you enter a password.  Only if something is invalid in the form would the password be set to what you used in a previous click to "Sign Up" button.  Even then it is a password you entered and it is set only to assist the user by not requiring them to reenter the same one they just typed.

Even when Google authentication is used to sign up a user, the password simply doesn't exist in our database.  Sign up with Google authentication leads to the user not being able to sign in without Google until they use the Password Change feature to set a new one.

**Comment by joshi1983** [comment]
@brucelau-github was assigned because he wanted to work on the email verification process.

This will likely involve:
- Adding a field to the user table for email_verification_token.
- Adding a field to the user table for email_verification_time
- Sending email when signing up for a new user account without third party authentication(Not applicable Google or Facebook signups)
- Handling a click to the emailed link by marking the email_verification_time to now

**Comment by brucelau-github** [comment]
Yeah. I think it is done month ago for this feature.

**Comment by Blandine-AA** [comment]
@joshi1983 @brucelau-github, email verification was done, but doesn't work properly. I opened an issue on it and asked @brucelau-github if he had time to look into fixing it.

> " what do you mean by "pre-populated"?
can be disregarded.

**Comment by joshi1983** [comment]
@Blandine-AA, @brucelau-github's changes were merged last night and put on both the demo and app.accesslocator.com.  That should finish the last 2 subtasks for this issue where signing in is blocked until the email address is confirmed.

I had a little difficulty with the emails getting in SPAM/JUNK folders so some people may face that.  That's the only problem I found while testing this latest change.  Do you think a new issue for the email to junk problem should be made instead of keeping this issue open?

**Comment by Blandine-AA** [comment]
@joshi1983, @brucelau-github, if work is completed then it should be set to "Ready for Testing". 

Also, because it was scheduled for 0.0.3, demo release shouldn't have been updated, without validating the change of release. Now either we undo the change to the demo release or we move this issue to 0.0.2 and update the release notes. Hoping that the test PASSES.

**Comment by joshi1983** [comment]
ok, I'll work on what @Blandine-AA said.  I'll do whichever is quickest and updating release notes looks like extra work for Areeba so I'll look into removing the updates from demo if that can be done with less effort.

**Comment by Blandine-AA** [comment]
@joshi1983, from the RN peerspective, it will be just adding and extra line for @areebakamal, but the feature needs to be tested and it has to be assigned to someone to do it. If after your assessment it is better keeping it in 0.0.2, then assign the testing to @kimficara. She will be faster at completing it by Friday night.

**Comment by Blandine-AA** [comment]
@brucelau-github, To help speeding up the testing, please provide some high level steps for testing this fix.
I know I have describe the problem in #199.

**Comment by joshi1983** [comment]
I deployed a change to undo the check for a verified email address when signing in.  I know there's a problem with some emails going to SPAM or junk folders so undoing the change on demo seemed better than getting the other testers to experience that problem.  This will also ease up things on Areeba and the rest of the team who are struggling to complete the testing and presentation site updates for the 0.0.2 release.

**Comment by joshi1983** [comment]
There is a lingering problem where emails go to SPAM sometimes but that's covered by #520 covers it.  I'm closing this because it is complete.
